Upgrade Django Q

tomijarvi opened the pull request:
The Django Q hasn't been stable in the servers and currently, async reports don't work anymore. The latest version contains some fixes which could help to this. This has been tested locally and it works now correctly so it should work in the servers as well.

suutari-ai commented:
If there was a problem with the old version, I think it should deserve a test case which verifies that the problem is now fixed and stays so in future updates too.

suutari-ai commented:
Please share all the relevant details in the commit message. That's important for the info to be available in the Git history. Currently there's more info in the PR description than in the commit message (it should be the other way around IMHO, or the same info in both).

suutari-ai commented:
It's very hard to see which packages got upgraded, since you're update the file format (new pip-tools version) in the same commit with the version upgrades. Maybe if you update pip-tools first and then just regenerate the txt files with new pip-tools without making any version changes, then in another commit do the version update. That way the first commit would contain just formatting changes and the version updates in the second commit would be easy to review.

tomijarvi commented:
@suutari-ai 
> If there was a problem with the old version, I think it should deserve a test case which verifies that the problem is now fixed and stays so in future updates too.

I added a simple test. It tests that the async report will be generated and it makes sure that the email has been set to outbox.

> It's very hard to see which packages got updated, since you're updating the file format (new pip-tools version) in the same commit with the version updates. Maybe if you update pip-tools first and just regenerate the txt files with new pip-tools without making any version changes. Then, in another commit, do the version update. That way the first commit would contain just formatting changes and the version updates in the second commit would be easy to review.

I made a new commit for the requirement's style changes.

suutari-ai reviewed (approved):
Good work Tomi! Thanks! :+1: 

There seems to be some issue in the test suite, but as far as I understand, it's unrelated to this PR.
